### ISSUE TITLE
`-ar` flags do not seem to work if they come before the filename.

### ISSUE THREAD

**Issue by janaakhterov** [sitkevij/hex#22]
```bash
~
❯ touch test

~
❯ echo "Hello, world." > test

~
❯ time hx -ar test
^CCommand terminated by signal 2
0.00user 0.00system 0:04.46elapsed 0%CPU (0avgtext+0avgdata 2528maxresident)k
0inputs+0outputs (0major+107minor)pagefaults 0swaps

~ took 4s
❯ time hx test -ar
let ARRAY: [u8; 14] = [
    0x48, 0x65, 0x6c, 0x6c, 0x6f, 0x2c, 0x20, 0x77, 0x6f, 0x72,
    0x6c, 0x64, 0x2e, 0x0a
];
0.00user 0.00system 0:00.00elapsed 93%CPU (0avgtext+0avgdata 2796maxresident)k
0inputs+0outputs (0major+112minor)pagefaults 0swaps

~```

**Comment by rudock1** [comment]
ditto.  

While help says hx [OPTIONS] [INPUTFILE], to get the array feature to work, order needs to be hx [INPUTFILE] [OPTIONS]
Otherwise, really cool hex dump tool.  I like it!

**Comment by poliorcetics** [comment]
I think #28 will fix this.

It seems to me this was the same bug as for `--len`, and with the changes in #28 it works for me:

```shell
# With the patch introduced in #28 applied.
❯ ./target/debug/hx -ar test
let ARRAY: [u8; 14] = [
    0x48, 0x65, 0x6c, 0x6c, 0x6f, 0x2c, 0x20, 0x77, 0x6f, 0x72, 
    0x6c, 0x64, 0x2e, 0x0a
];
```

**Comment by sitkevij** [comment]
Closing out. Resolved as of release v0.3.1 https://github.com/sitkevij/hex/releases/tag/v0.3.1
